eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/OrderedMapAdapter.java

```diff
@@ -571,9 +571,8 @@ public <VV> MutableListMultimap<VV, V> groupByEach(Function<? super V, ? extends
     @Override
     public <VV> MutableOrderedMap<VV, V> groupByUniqueKey(Function<? super V, ? extends VV> function)
     {
-        // TODO: Make this work in a major version by changing {@link org.eclipse.collections.api.RichIterable#groupByUniqueKey}'s type R from MutableMap to MutableMapIterable.
-        // return this.groupByUniqueKey(function, this.newEmpty());
-        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".groupByUniqueKey() not implemented yet");
+        MutableOrderedMap<VV, V> vs = (MutableOrderedMap<VV, V>) this.newEmpty();
```

Comment:
Was the TODO incorrect?

Reply:
Yes, it was stale, I already fixed that return type a while ago.